### PR TITLE
Add volume control options

### PR DESCRIPTION
**Pull request by Dennisbonke**
Are YOU tired of going deaf every time you load the game and no way to change it? Fear no more! This PR brings state of the art volume control for both the background music and the sound effects to you! The revolutionary use of two separate sliders allows you to control both volumes independently of each other.

### CODE REVIEW

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+			optionsMenu_.SetActive(true);
+		}
+
+		private void UpdateSliders() {
+			float backgroundMusicVolume;
+			backgroundMusicMixer_.GetFloat("MusicVolume", out backgroundMusicVolume);
+			backgroundMusicSlider_.value = Mathf.Pow(10, backgroundMusicVolume / 20);
+
+			float soundEffectVolume;
+			soundEffectMixer_.GetFloat("MusicVolume", out soundEffectVolume);
```
Comment (Westlanderz) [inline Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:30]: What is this string in this function `MusicVolume`? And why is it the same for this and the background music slider?

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+namespace OperationBlackwell.UI {
+	public class AudioOptionsMenu : Menu {
+
+		[SerializeField] private Slider backgroundMusicSlider_;
+		[SerializeField] private Slider soundEffectSlider_;
+		[SerializeField] private AudioMixer backgroundMusicMixer_;
+		[SerializeField] private AudioMixer soundEffectMixer_;
+
+		private void Start() {
+			UpdateSliders();
```
Comment (Westlanderz) [inline Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:16]: Are the Sliders only updated on start??

File: Assets/Scripts/Core/Utils/VolumeControl.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Audio;
+
+namespace OperationBlackwell.Core {
+    public class VolumeControl : MonoBehaviour {
```
Comment (Westlanderz) [inline Assets/Scripts/Core/Utils/VolumeControl.cs:6]: What does this class do?

File: Assets/Scripts/UI/Menus/OptionsMenu.cs [hunk 4]
```diff
@@ -4,10 +4,6 @@
 
 namespace OperationBlackwell.UI {
 	public class OptionsMenu : Menu {
-		public void AudioClicked() {
```
Comment (Westlanderz) [inline Assets/Scripts/UI/Menus/OptionsMenu.cs:7]: Was this only removed, and how do we enter the audio menu now?

**Review by Westlanderz — APPROVED** [review]
I saw not code issues, i only have some questions about the code just for explination.

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:30]: It's the name of the exposed variable from the editor. As both sliders refer to their own mixer, having them have the same name isn't an issue.

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs (hunk 2 above)
Comment (Dennisbonke) [reply Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:16]: Nope, but this forces them to be whatever the current value the mixer has on scene load.

File: Assets/Scripts/UI/Menus/OptionsMenu.cs (hunk 4 above)
Comment (Dennisbonke) [reply Assets/Scripts/UI/Menus/OptionsMenu.cs:7]: It was a placeholder, we now set the correct gameobject to active and the main options menu to inactive, but that happens with events and is set up within the editor.

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs (hunk 2 above)
Comment (Westlanderz) [reply Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:16]: Okay, but how are the sliders updated then?

Comment (Dennisbonke) [reply Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:16]: By moving them? It's just an unity slider.

File: Assets/Scripts/Core/Utils/VolumeControl.cs (hunk 3 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Utils/VolumeControl.cs:6]: So this is used as a seperate component on the sliders? Should this not be directly integrated into the sliders directly?

Comment (Dennisbonke) [reply Assets/Scripts/Core/Utils/VolumeControl.cs:6]: I can't add it directly in the sliders as the sliders are unity and I can't edit those? I'm not quite sure what you mean here.

File: Assets/Scripts/UI/Menus/AudioOptionsMenu.cs (hunk 2 above)
Comment (Westlanderz) [reply Assets/Scripts/UI/Menus/AudioOptionsMenu.cs:16]: Im just stupid leave me alone........

File: Assets/Scripts/Core/Utils/VolumeControl.cs (hunk 3 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/Utils/VolumeControl.cs:6]: No never mind, what i had in thought does not work, as the updating of the mixer value happens differently then i thought.

**Pull request merged**